src/queens/distributions/beta.py

```diff
@@ -30,108 +31,108 @@ class Beta(Continuous):
-            upper_bound (np.ndarray): Upper bound of the beta distribution.
-            a (float): Shape parameter of the beta distribution, must be > 0.
-            b (float): Shape parameter of the beta distribution, must be > 0.
+            lower_bound: Lower bound of the beta distribution.
+            upper_bound: Upper bound of the beta distribution.
+            a: Shape parameter of the beta distribution, must be > 0.
+            b: Shape parameter of the beta distribution, must be > 0.
         """
         lower_bound = np.array(lower_bound).reshape(-1)
         upper_bound = np.array(upper_bound).reshape(-1)
-        a = np.array(a)
-        b = np.array(b)
-        super().check_positivity(a=a, b=b)
+        self.a = np.array(a)
+        self.b = np.array(b)
+        super().check_positivity(a=self.a, b=self.b)
         super().check_bounds(lower_bound, upper_bound)
         scale = upper_bound - lower_bound
-        scipy_beta = scipy.stats.beta(scale=scale, loc=lower_bound, a=a, b=b)
+        scipy_beta = scipy.stats.beta(scale=scale, loc=lower_bound, a=self.a, b=self.b)
         mean = scipy_beta.mean()
         covariance = scipy_beta.var()
 
         self.lower_bound = lower_bound
         self.upper_bound = upper_bound
-        self.a = a
-        self.b = b
         self.scipy_beta = scipy_beta
 
         super().__init__(mean=mean, covariance=covariance, dimension=1)
 
-    def cdf(self, x):
+    def cdf(self, x: np.ndarray) -> np.ndarray:
         """Cumulative distribution function.
 
         Args:
-            x (np.ndarray): Positions at which the *cdf* is evaluated
+            x: Positions at which the CDF is evaluated
 
         Returns:
-            cdf (np.ndarray): CDF at evaluated positions
+            CDF at positions
         """
         cdf = self.scipy_beta.cdf(x).reshape(-1)
         return cdf
 
-    def draw(self, num_draws=1):
+    def draw(self, num_draws: int = 1) -> np.ndarray:
         """Draw samples.
 
         Args:
-            num_draws (int, optional): Number of draws
+            num_draws: Number of draws
 
         Returns:
-            samples (np.ndarray): drawn samples from the distribution
+            Drawn samples from the distribution
         """
         samples = self.scipy_beta.rvs(size=num_draws).reshape(-1, 1)
         return samples
 
-    def logpdf(self, x):
+    def logpdf(self, x: np.ndarray) -> np.ndarray:
         """Log of the probability density function.
 
         Args:
-            x (np.ndarray): Positions at which the log pdf is evaluated
+            x: Positions at which the log-PDF is evaluated
 
         Returns:
-            logpdf (np.ndarray): log pdf at evaluated positions
+            Log-PDF at positions
         """
         logpdf = self.scipy_beta.logpdf(x).reshape(-1)
         return logpdf
 
-    def grad_logpdf(self, x):
-        """Gradient of the log pdf with respect to *x*.
+    def grad_logpdf(self, x: np.ndarray) -> Any:
```

Comment:
Technically 
```suggestion
    def grad_logpdf(self, x: np.ndarray) -> None:
```
as this method is not implemented...